service/history/queues/executable.go

```diff
@@ -524,51 +524,38 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 			// if err is due to workflow busy or APS limit, do not take any latency related to this attempt into account
 			e.inMemoryNoUserLatency += e.scheduleLatency + e.attemptNoUserLatency
 		}
-
-		if retErr != nil {
-			e.Lock()
-			defer e.Unlock()
-
-			e.attempt++
-			if e.attempt > taskCriticalLogMetricAttempts {
-				metrics.TaskAttempt.With(e.metricsHandler).Record(int64(e.attempt))
-				e.logger.Error("Critical error processing task, retrying.",
-					tag.Attempt(int32(e.attempt)),
-					tag.UnexpectedErrorAttempts(int32(e.unexpectedErrorAttempts)),
-					tag.Error(err),
-					tag.OperationCritical)
-			}
-		}
 	}()
 
-	if len(e.dlqErrorPattern()) > 0 {
-		match, mErr := regexp.MatchString(e.dlqErrorPattern(), err.Error())
-		if mErr != nil {
-			e.logger.Error(fmt.Sprintf("Failed to match task processing error with %s", dynamicconfig.HistoryTaskDLQErrorPattern.Key()))
-		} else if match {
-			e.logger.Error(
-				fmt.Sprintf("Error matches with %s. Marking task as terminally failed, will send to DLQ",
-					dynamicconfig.HistoryTaskDLQErrorPattern.Key()),
-				tag.Error(err),
-				tag.ErrorType(err))
-			e.terminalFailureCause = err
-			metrics.TaskTerminalFailures.With(e.metricsHandler).Record(1)
-			return fmt.Errorf("%w: %v", ErrTerminalTaskFailure, err)
-		}
+	if matchedErr := e.matchDLQErrorPattern(err); matchedErr != nil {
```

Comment:
This internally emits a log, so the logging below isn't really necessary. 

We don't log "Fail to process task" before the change as well.